compiler/rustc_codegen_llvm/src/va_arg.rs

```diff
@@ -218,6 +237,148 @@ fn emit_aapcs_va_arg<'ll, 'tcx>(
     val
 }
 
+fn emit_powerpc_va_arg<'ll, 'tcx>(
+    bx: &mut Builder<'_, 'll, 'tcx>,
+    list: OperandRef<'tcx, &'ll Value>,
+    target_ty: Ty<'tcx>,
+) -> &'ll Value {
+    let dl = bx.cx.data_layout();
+
+    // struct __va_list_tag {
+    //   unsigned char gpr;
+    //   unsigned char fpr;
+    //   unsigned short reserved;
+    //   void *overflow_arg_area;
+    //   void *reg_save_area;
+    // };
+    let va_list_addr = list.immediate();
+
+    // Peel off any newtype wrappers.
+    let layout = {
+        let mut layout = bx.cx.layout_of(target_ty);
+
+        while let Some((_, inner)) = layout.non_1zst_field(bx.cx) {
+            layout = inner;
+        }
+
+        layout
+    };
+
+    // Rust does not support any powerpc softfloat targets.
+    let is_soft_float_abi = false;
```

Comment:
for our aarch64 and loongarch64 softfloat targets they have "softfloat" as an ABI string

can we do
```rust
let is_soft_float_abi = target.abi == "softfloat";
// Rust does not support any powerpc softfloat targets.
assert!(!is_soft_float_abi);
```

Reply:
variant spellings of the same idea are fine too